microfs.py

```diff
@@ -56,10 +56,16 @@ def raw_on(serial):
     """
     Puts the device into raw mode.
     """
-    serial.write(b'\x03')  # Send CTRL-C to break out of loop.
-    serial.read_until(b'\n>')  # Flush buffer until prompt.
-    serial.write(b'\x01')  # Go into raw mode.
-    serial.read_until(b'\r\n>OK')  # Flush buffer until raw mode prompt.
+    attempts_left = 3
+    while attempts_left:
+        serial.write(b'\x03')    # Send CTRL-C to break out of loop.
+        if serial.read_until(b'\n>>>').endswith(b'\n>>>'):
```

Comment:
I think the problem is that if you are already in raw REPL then ctrl-C won't yield `>>>`, but rather something else (like just `>`).

Reply:
Yeah, that's a good point. It'll be better to leave the original `b'\n>'`.